bio-distance/src/main/kotlin/bio/relatives/bio/distance/parser/BEDFeatureParser.kt

```diff
@@ -0,0 +1,58 @@
+package bio.relatives.bio.distance.parser
+
+import bio.relatives.bio.distance.model.BEDFeature
+import bio.relatives.common.model.Feature
+import bio.relatives.common.parser.FeatureParser
+import bio.relatives.common.parser.FeatureParser.Companion.FEATURE_FILE_EXTENSION
+import bio.relatives.common.parser.FeatureParser.Companion.MAX_FEATURE_SIZE
+import bio.relatives.common.utils.isValid
+import org.springframework.stereotype.Component
+import java.nio.file.Path
+
+/**
+ * @author Created by Vladislav Marchenko on 02.02.2021
+ */
+@Component
+class BEDFeatureParser : FeatureParser {
+    override fun parseFeatures(featureFilePath: Path): List<Feature> {
+        featureFilePath.isValid(requiredExtension = FEATURE_FILE_EXTENSION)
+
+        val res = mutableListOf<BEDFeature>()
+
+        featureFilePath.toFile().forEachLine {
+            run {
+                if (it != "/n" && !it.startsWith("#")) {
+                    val rows: Array<String> = it.split("[\\s+\\t+]".toRegex()).filter { it.isNotBlank() }.toTypedArray()
+
+                    require(rows.size == 4) {
```

Comment:
in `abstract fun process(rows: List<String>): Feature`

Reply:
completed